libmdnsd/CMakeLists.txt

```diff
@@ -63,13 +67,33 @@ if(MDNSD_COMPILE_AS_CXX)
 endif()
 
 if(MDNSD_BUILD_OSS_FUZZ)
-    add_library(libmdnsd ${LIBRARY_FILES})
+    add_library(libmdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
 else()
-    add_library(libmdnsd SHARED ${LIBRARY_FILES})
+    add_library(libmdnsd SHARED ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
```

Comment:
Why always build a shared library?
This can be modified externally, by `BUILD_SHARED_LIBS`.